spring-cloud-function-adapters/spring-cloud-function-grpc/src/main/java/org/springframework/cloud/function/grpc/GrpcServer.java

```diff
@@ -70,7 +78,12 @@ public void start() {
 
 				logger.info("Starting gRPC server");
 				this.server.start();
-				logger.info("gRPC server is listening on port " + this.grpcProperties.getPort());
+				logger.info("gRPC server is listening on port " + this.server.getPort());
+
+				if (environment instanceof ConfigurableEnvironment) {
+					((ConfigurableEnvironment) this.environment).getPropertySources().addFirst(
+						new MapPropertySource("grpcServerProps", Collections.singletonMap("local.grpc.server.port", server.getPort())));
```

Comment:
This is actual a feature improvement as it will allow others to get at the local grpc port (same as is available for Web and RSocket)